### PR TITLE
Embed descriptors to work-around reflection issues with health and server reflection services

### PR DESCRIPTION
**Pull request by jhump**
This is an alternate (and much more concise) implementation over #48. The first four commits actually show the alternate approach, and the subsequent two commits mostly delete stuff from the first few 😂.

### CODE REVIEW

File: grpcreflect.go [hunk 1]
```diff
@@ -408,74 +392,31 @@ func (n *staticNames) Names() []string {
...
-			// we need to create "unmangled" versions of the services
-			// that connect libs have to mangle
-			res = r.doInit(res)
-			desc, err = res.FindDescriptorByName(name)
-			if err == nil {
-				return desc, nil
-			}
-		}
-		return nil, err
+		return resolver
```
Comment (akshayjshah) [inline grpcreflect.go:414]: I'm uncomfortable ignoring errors here - don't they indicate some critical incompatibility between the embedded binary data, the schemas, and `protodesc`? If the owner of `main` is relying on reflection to find the health and reflection services, we don't want to fail at runtime with no explanation.

Could we instead return `(protodesc.Resolver, error)`, capture the error as a global, and return it from `combinedResolver` if we ever need to look up symbols in `.second`?

File: Makefile [hunk 2]
```diff
@@ -89,3 +89,10 @@ $(BIN)/golangci-lint: Makefile
 $(BIN)/protoc-gen-go: Makefile
 	@mkdir -p $(@D)
 	GOBIN=$(abspath $(@D)) $(GO) install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28.1
+
+services.bin: $(BIN)/buf
+	$(BIN)/buf build --as-file-descriptor-set --output $(@F) \
+		buf.build/grpc/grpc:26635376b3f47a11126a0f4b4b5b6de7fe5a074a \
+		--type grpc.health.v1.Health \
+		--type grpc.reflection.v1.ServerReflection \
+		--type grpc.reflection.v1alpha.ServerReflection
```
Comment (akshayjshah) [inline Makefile:98]: nit: can we keep a trailing newline here, so `cat Makefile` displays properly?

**Review by akshayjshah — APPROVED** [review]

File: grpcreflect.go (hunk 1 above)
Comment (jhump) [reply grpcreflect.go:414]: > don't they indicate some critical incompatibility between the embedded binary data, the schemas, and protodesc?

Yes, but it's also a condition that would certainly be caught during CI testing.

> Could we instead return `(protodesc.Resolver, error)`, capture the error as a global, and return it from `combinedResolver` if we ever need to look up symbols in `.second`?

Sure.

Comment (jhump) [reply grpcreflect.go]: Done. 0df528356d1aca3c1b57b7cb9da1b4d2f907016e

File: Makefile (hunk 2 above)
Comment (jhump) [reply Makefile]: Done. 0df528356d1aca3c1b57b7cb9da1b4d2f907016e

**Pull request merged**